python/tank/platform/constants.py

```diff
@@ -175,6 +175,12 @@
 # the name of the shell engine
 SHELL_ENGINE = "tk-shell"
 
+# valid characters for a template key name
+TEMPLATE_KEY_NAME_REGEX = "[a-zA-Z_ 0-9\.]+"
+
+# a human readable explanation of the above. For error messages.
+VALID_TEMPLATE_KEY_NAME_DESC = "letters, numbers, underscore, space and period"
```

Comment:
It's following the pattern here: https://github.com/shotgunsoftware/tk-core/blob/27823_unregister_folders/python/tank/platform/constants.py#L130

Seems odd to define the constant in one place and then have error messages that reflect the value of that constant in another place. Very easy to forget to update one when the other is changed?